### DISCUSSION TITLE
how would be a ssrExchange for multiple pages

### DISCUSSION THREAD

**Discussion by edw19** [urql-graphql/urql#2483] [Q&A] [answered]
I am configuring withUrqlClient in my app but it is costing me a lot of work to be able to share the ssrExchange to share the queries in getServerSideProps as. Is there an optimal way to do this?

**Comment by edw19** [comment]
i'm currently running this

it FUNCION
but I have to press f5 twice and it detects the changes made in the cache

![image](https://user-images.githubusercontent.com/47750859/172494108-508ea602-cf75-42d5-afeb-033ac66945b1.png)

**Comment by JoviDeCroock** [comment, marked as answer]
This already functions for multiple pages, i.e. all the fetches you do are added as keys and can then be rehydrated from [here](https://github.com/FormidableLabs/urql/blob/main/packages/next-urql/src/with-urql-client.ts#L63-L75) the else branch here is for client-side transitions that introduce new server-orchestrated state
